### DISCUSSION TITLE
How to use generated output as the body of an outer code block?

### DISCUSSION THREAD

**Discussion by sisp** [pawamoy/markdown-exec#38] [Q&A] [answered]
I'm trying to generate content for a fenced code block and thought `markdown-exec` could do just that, but I'm hitting a wall.

I'd like to render a Vega-Lite chart in MkDocs using [`mkdocs-charts-plugin`](https://github.com/timvink/mkdocs-charts-plugin). But instead of hardcoding the Vega-Lite JSON document, I'd like to generate it, e.g. using some inline Python code or using a shell command. Before actually trying it out and seeing it fail, I thought this was going to work:

`````md
````vegalite
```python exec="1"
# Imagine a real function that generates a Vega-Lite JSON document
print('{"$schema": "https://vega.github.io/schema/vega-lite/v5.json", ...}')
```
````
`````

But `mkdocs-charts-plugin` fails to render the Vega-Lite JSON document because it receives literally this

````
```python exec="1"
# Imagine a real function that generates a Vega-Lite JSON document
print('{"$schema": "https://vega.github.io/schema/vega-lite/v5.json", ...}')
```
````

instead of the result from executing this Python code.

After thinking about this problem a little, it probably wouldn't be a good idea – if technically possible at all – to always render inner code blocks with `exec="1"` and pass the output to the outer code block because then writing documentation that shows a code block with `exec="1"` wouldn't be possible (it would be rendered and not left untouched).

That said, is there any way to do what I'm trying to achieve, ideally in a generic way without needing to add explicit support for the `vegalite` fence (which I could do of course – if you accepted the contribution –, but it wouldn't scale to other similar use cases)?

**Comment by pawamoy** [comment, marked as answer]
Hey @sisp, happy to see you here :smile: 

So, yeah, your reasoning is correct, we can't execute inner code blocks if the outer is not also executed. We cannot even *detect* that the inner block should be executed: we rely on the PyMDown Superfences extension (not regular expressions or other similar approaches), so we only ever see outer code blocks: the inner code blocks are just seen as contents of the outer code blocks.

Now, I believe what you want to achieve is possible. Try this:

````
```python exec="1" result="vegalite"
# Imagine a real function that generates a Vega-Lite JSON document
print('{"$schema": "https://vega.github.io/schema/vega-lite/v5.json", ...}')
```
````

This simply tells Markdown Exec to wrap the output of your executed code block in a `vegalite` code block, and render it again from Markdown to HTML :slightly_smiling_face:

**Reply by sisp** [reply]
OMG, I can't believe I missed this option in the docs. :facepalm: It's the solution to my problem. :ok_hand: Sorry for the bother because of my not reading the docs properly. 🫣 `markdown-exec` is such an amazing MkDocs plugin! :rocket:

**Reply by pawamoy** [reply]
No worries haha, and thanks :heart: !
